crates/ruff_python_formatter/src/builders.rs

```diff
@@ -82,79 +74,32 @@ impl<'buf, 'ast> PyFormatterExtensions<'ast, 'buf> for PyFormatter<'ast, 'buf> {
 pub(crate) struct JoinNodesBuilder<'fmt, 'ast, 'buf> {
     fmt: &'fmt mut PyFormatter<'ast, 'buf>,
     result: FormatResult<()>,
-    last_end: Option<TextSize>,
-    node_level: NodeLevel,
 }
 
 impl<'fmt, 'ast, 'buf> JoinNodesBuilder<'fmt, 'ast, 'buf> {
```

Comment:
This now has nothing specific to AST nodes (if that's what "nodes" means here), instead it's just a joiner that requires you to specify a custom separator for every node (which the generic joiner doesn't support).